src/background/dapp-session.manager.ts

```diff
@@ -56,19 +85,45 @@ class DappSecurityContext {
   }
 }
 
+class ExtensionDappSecurityContext extends DappSecurityContext {
+  constructor() {
+    super('extension')
```

Comment:
this will mix two different and independent extension's context. 
it should be distinguished by their extension id or only the blossom extension should be able to call the `swarm-extension`

Reply:
The `ExtensionDappSecurityContext` is actually stateless. If in the future we want to store some state, then we can use extension ID to distinguish different extensions.